### PR TITLE
Nix Shell Segment

### PR DESCRIPTION
**Pull request by yash-garg**
### Prerequisites

- [x] I have read and understood the [contributing guide][CONTRIBUTING.md].
- [x] The commit message follows the [conventional commits][cc] guidelines.
- [x] Tests for the changes have been added (for bug fixes / features).
- [x] Docs have been added/updated (for bug fixes / features).

### Description

This adds a segment for [Nix Shell](https://nixos.org/guides/nix-pills/developing-with-nix-shell.html).

Please review and let me know if there are any improvements needed in the code or documentation.

[CONTRIBUTING.md]: https://github.com/JanDeDobbeleer/oh-my-posh/blob/main/CONTRIBUTING.md
[cc]: https://www.conventionalcommits.org/en/v1.0.0/#summary

### CODE REVIEW

File: src/config/segment_types.go [hunk 1]
```diff
@@ -128,6 +128,8 @@ const (
 	NPM SegmentType = "npm"
 	// NX writes which Nx version us currently active
 	NX SegmentType = "nx"
+	// NIX writes the active nix shell details
+	NixShell SegmentType = "nix_shell"
```
Comment (JanDeDobbeleer) [inline src/config/segment_types.go]:
```suggestion
	NIX SegmentType = "nix"
```

File: src/segments/nixshell.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	default:
+		if n.InNewNixShell() {
+			return UNKNOWN
+		}
+		return NONE
+	}
+}
+
+// Hack to detect if we're in a `nix shell` (in contrast to a `nix-shell`).
+func (n *NixShell) InNewNixShell() bool {
```
Comment (JanDeDobbeleer) [inline src/segments/nixshell.go:44]: I don't think this offer any guarantees, it assumes the availability of one `PATH` entry implies we're in a nix shell whereas you could still have, and be running, `nix-shell` if that one's also on your `PATH`. Also, why is this relevant? If the shell type is there (`IN_NIX_SHELL`), do you need to do this validation?

Comment (yash-garg) [reply src/segments/nixshell.go:44]: @JanDeDobbeleer so there are two types of shells by design, `nix shell` and `nix-shell`, some discussion can be found [here](https://discourse.nixos.org/t/nix-shell-nix-shell-and-nix-develop/25964).

When we use `nix shell` the `IN_NIX_SHELL` variable isn't set, see https://github.com/NixOS/nix/issues/6677 so we check the `PATH` for it to verify.

https://starship.rs/ also uses the same approach for detecting it.

Comment (JanDeDobbeleer) [reply src/segments/nixshell.go:44]: @Yash-Garg can you add that issue to the inline comments as well. I don't understand how validating a PATH entry even closely qualifies as implying being in a certain shell environment, but I guess that's just me.

Comment (yash-garg) [reply src/segments/nixshell.go:44]: @JanDeDobbeleer sure, i've added the comments for it.

**Comment by JanDeDobbeleer** [comment]
@all-contributors please add @Yash-Garg for code,design,doc

This is pure magic 🪄! @holopin-bot @Yash-Garg wizard

**Pull request merged**